insights/parsers/engine_config.py

```diff
@@ -0,0 +1,187 @@
+in output using ``__contains__``::
+
+    >>> 'MaxRerunVmOnVdsCount' in output
+    True
+
+A value of keyword can be retrieved using ``__getitem__``::
+
+    >>> output['MaxRerunVmOnVdsCount']
+    ['3']
+
+The values will be in a form of list as there can be keyword with same
+name and different values::
+
+    >>> output['HotUnplugCpuSupported']
+    ['{"x86":"false","ppc":"false"}', '{"x86":"false","ppc":"false"}', '{"x86":"true","ppc":"true"}']
+
+If `value` is empty or nil an empty list is returned::
+
+    >>> output['ClusterRequiredRngSourcesDefault']
+    []
+
+``head(keyword)`` & ``last(keyword)`` can be use to retrieve first and last value from values::
+
+    >>> output.head('HotUnplugCpuSupported')
+    '{"x86":"false","ppc":"false"}'
+    >>> output.last('HotUnplugCpuSupported')
+    '{"x86":"true","ppc":"true"}'
+
+``get_version(keyword)`` can be used to retrieve versions(list)::
+
+    >>> output.get_version('HotUnplugCpuSupported')
+    ['3.6', '4.0', '4.1']
+"""
+from collections import namedtuple
+from operator import itemgetter
+from .. import Parser, parser, get_active_lines
+
+
+@parser("engine-config_all")
+class EngineConfigAll(Parser):
+    """Parsing output of command ``engine-config --all``
+
+    The parser tries its best to get `value` & `version` for `keyword`
+    specified. It ignores keywords where is fails. At the moment it
+    works well for output which has `keyword`, `value` & `version` in
+    a single line. It skip(rather fails) any `keyword` which has
+    multi-line output.
+
+    Attributes:
+        fields (list): List of `KeyValue` namedtupules for
+                       each line in the configuration file.
+
+        keywords (set): Set of keywords present in the configuration
+                        file, each keyword has been converted to lowercase.
+    """
+    keyvalue = namedtuple('KeyValue',
+                          ['keyword', 'value', 'version', 'kw_lower'])
+    """namedtuple: Represent name value pair as a namedtuple with case."""
+
+    def parse_content(self, content):
+        """Parse each active line for keyword, values & version
+
+        Args:
+
+            content (list): Output of command ``engine-config --all``
+        """
+        self.fields = []
+        for line in get_active_lines(content):
+            try:
+                key, val, ver = itemgetter(0, 1, -1)(line.split(' '))
+                self.fields.append(self.keyvalue(key.strip(':'), val, ver, key.strip(':').lower()))  # noqa
+            except:
+                pass
+        self.keywords = set([kw.kw_lower for kw in self.fields])
+
+    def __contains__(self, keyword):
+        return keyword.lower() in self.keywords
+
+    def __iter__(self):
+        return iter(self.fields)
+
+    def __getitem__(self, keyword):
+        kw = keyword.lower()
+        if kw in self.keywords:
+            return filter(
+                None,
+                [kv.value for kv in self.fields if kv.kw_lower == kw])
```

Comment:
@psachin I noticed that in your example code above at line 44 an empty list is returned but here if `keyword` is not in `self.keywords` then `None` is returned.  Should you return an empty list instead?  Also could you could simplify this return as just `return [kv.value for kv in self.fields if kv.kw_lower == kw]` or is there some reason why `filter` is necessary?

Reply:
Hi @bfahr I use `filter(None, ...)` on purpose. As you see in the example, the keyword 'ClusterRequiredRngSourcesDefault' has empty values(returns `['', '', '']`). I want `__getitem__` to return empty list in case all values in list are empty(now documented). 

The current implementation returns `False` if keyword don't exists but values and versions will be empty list for non-existing keyword(documented). 

Another approach is to keep the list as it is(`['', '', '']`) rather then returning `[]` and map them to versions(say `['3.0', '3.6', '4.0']`) so that user can fetch values specific to version(or vice versa) for a keyword. But I think that is not needed at this time and I don't have a use case. 

Let me know you view